lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -538,6 +543,28 @@ func makeSubscriberStateHandler(desiredState string) echo.HandlerFunc {
 	}
 }
 
+func getStateExactIMSIPrefix(networkID string, typeFilter []string, keyFilter []string, keyPrefix *string, serdes serde.Registry) (state_types.StatesByID, error) {
+	// post-filter functionality to only return states with the exact IMSI prefix as specified
+	states, err := state.SearchStates(networkID, typeFilter, keyFilter, keyPrefix, serdes)
+	if err != nil {
+		return nil, err
+	}
+
+	if !funk.IsEmpty(keyPrefix) {
```

Comment:
This function is only ever used for the purpose of using keyPrefix, so rework this so that you don't need this check